File: kb-core-ui/harness/tests/fixtures/typescript-basics/repo/mathutil.ts
/** Starting total for a Calculator. */
export const DEFAULT_START = 0;

/** Returns the sum of two numbers. */
export function add(a: number, b: number): number {
  return a + b;
}

/** Something that can add a number to a running total. */
export interface Adder {
  addTo(n: number): number;
}

/** Accumulates a running total. */
export class Calculator implements Adder {
  total: number = DEFAULT_START;

  addTo(n: number): number {
    this.total = add(this.total, n);
    return this.total;
  }
}

File: kb-core-ui/harness/tests/fixtures/go-basics/repo/mathutil.go
package mathutil

// DefaultStart is the calculator's starting total.
const DefaultStart = 0

// Add returns the sum of two integers.
func Add(a, b int) int {
	return a + b
}

// Adder is implemented by anything that can add.
type Adder interface {
	AddTo(n int) int
}

// Calculator accumulates a running total.
type Calculator struct {
	Total int
}

// AddTo adds n to the calculator's total and returns the new total.
func (c *Calculator) AddTo(n int) int {
	c.Total = Add(c.Total, n)
	return c.Total
}
